trino/sqlalchemy/dialect.py

```diff
@@ -106,6 +111,10 @@ def create_connect_args(self, url: URL) -> Tuple[Sequence[Any], Mapping[str, Any
             kwargs["http_scheme"] = "https"
             kwargs["auth"] = CertificateAuthentication(unquote_plus(url.query['cert']), unquote_plus(url.query['key']))
 
+        if "externalAuthentication" in url.query:
```

Comment:
That's what I've been doing when using the client directly. But I've been running into an issue passing it as a class when trying to use it from some other projects such as [ipython-sql](https://github.com/catherinedevlin/ipython-sql).
Eventually I found out that `ipython-sql` allows you to actually pass the `-creator` parameter to bypass the configuration by sending a connection straight away but it'd be great if it would work without this workaround.

Reply:
Do we have good behaviour when someone has externalAuthentication in URI and has explicitly set a different auth mechanism? What takes precedence? I'd like to avoid this situation of having multiple ways of configuring something since it leads to more combination of things to test and edge cases to think about.

Reply:
Hey @hashhar, I have similar feelings to be honest. I just extended the way it's being done for SQLAlchemy URIs (It's already done this way for Basic Authentication, JWT Authentication and Cert Authentication).

I think it makes sense to align with the JDBC driver and have this for SQLAlchemy... If the user passes multiple Auth Methods I think it's on the user side to be honest, but I guess it could be improved by validating if they did this somehow (checking the query args and `connect_args`)

Reply:
This validation seems useful but out of scope of this PR because it needs to apply to other auth methods as outlined above.

Reply:
You mean validating if the user passes multiple auth methods?

So you think it'd be fine to keep that responsibility on the user side so far? I'd be fine with it (=